go/vt/discovery/tablet_picker_test.go

```diff
@@ -62,6 +62,33 @@ func TestPickPrimary(t *testing.T) {
 	assert.True(t, proto.Equal(want, tablet), "Pick: %v, want %v", tablet, want)
 }
 
+// TestPickNoPrimary confirms that if the picker was setup only for primary tablets but
+// there is no primary setup for the shard we correctly return an error.
+func TestPickNoPrimary(t *testing.T) {
+	defer utils.EnsureNoLeaks(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 200*time.Millisecond)
```

Comment:
Why use multiple contexts, and why with such short timeouts? Won't this make it flaky in the CI?

Reply:
It was a cut/paste of a previous test and I had only changed one line, so didn't focus on that. Changed to use a single context.